tracer/src/Datadog.Trace/Iast/Aspects/System.Text/StringBuilderAspects.cs

```diff
@@ -174,31 +192,680 @@ public static StringBuilder Append(StringBuilder target, object? value)
+        var result = target!.AppendJoin(separator, values!);
+        StringBuilderModuleImpl.FullTaintIfAnyTaintedEnumerable(target, separator, values);
+        return result;
+    }
+
+    /// <summary>StringBuilder.AppendJoin aspect</summary>
+    /// <param name="target">The StringBuilder instance.</param>
+    /// <param name="separator">The character to use as a separator.</param>
+    /// <param name="values">An array that contains the strings to concatenate and append to the current instance of the string builder.</param>
+    /// <returns>The modified StringBuilder instance.</returns>
+    [AspectMethodReplace("System.Text.StringBuilder::AppendJoin(System.Char,System.String[])")]
+    public static StringBuilder AppendJoin(StringBuilder? target, char separator, string[]? values)
+    {
+        // We want the null reference exception to be launched here if target is null
+        var result = target!.AppendJoin(separator, values!);
+        StringBuilderModuleImpl.FullTaintIfAnyTaintedEnumerable(target, null, values);
+        return result;
+    }
+
+    /// <summary>StringBuilder.AppendJoin aspect</summary>
+    /// <param name="target">The StringBuilder instance.</param>
+    /// <param name="separator">The character to use as a separator.</param>
+    /// <param name="values">An array that contains the strings to concatenate and append to the current instance of the string builder.</param>
+    /// <returns>The modified StringBuilder instance.</returns>
+    [AspectMethodReplace("System.Text.StringBuilder::AppendJoin(System.Char,System.Object[])")]
+    public static StringBuilder AppendJoin(StringBuilder? target, char separator, object[]? values)
+    {
+        // We want the null reference exception to be launched here if target is null
+        var result = target!.AppendJoin(separator, values!);
+        StringBuilderModuleImpl.FullTaintIfAnyTaintedEnumerable(target, null, values);
+        return result;
     }
+
+    /// <summary>StringBuilder.AppendJoin aspect</summary>
+    /// <param name="target">The StringBuilder instance.</param>
+    /// <param name="separator">The character to use as a separator.</param>
+    /// <param name="values">An array that contains the strings to concatenate and append to the current instance of the string builder.</param>
+    /// <returns>The modified StringBuilder instance.</returns>
+    [AspectMethodReplace("System.Text.StringBuilder::AppendJoin(System.Char,System.Collections.Generic.IEnumerable`1<!!0>)")]
+    public static StringBuilder AppendJoin(StringBuilder? target, char separator, IEnumerable? values)
+    {
+        if (values is null || target is null)
+        {
+#pragma warning disable CS8604 // We want to call the exact method overload even if values is null
+            return target!.AppendJoin(separator, values as IEnumerable<object>);
+#pragma warning restore CS8604 // Enable
+        }
+
+        var valuesConverted = values as IEnumerable<object>;
```

Comment:
Everithing appart from the call to the original method should be covered by a try catch. Although it's seemingly safe